core/src/smartcontracts/isi/permissions/combinators.rs

```diff
@@ -2,172 +2,103 @@
 
 use super::*;
```

Comment:
We need to get rid of this module in favour of a common combinatorial API.

Reply:
Well, maybe someday... I'll add a todo

Reply:
Is there an issue for common combinatorial API?